### ISSUE TITLE
Preview of a matrix with a variable used triggers "The evaluated answer is not an answer hash : ||." warning

### ISSUE THREAD

**Issue by taniwallach** [openwebwork/pg#619]
When a question uses a MathObject matrix to generate an `ans_array` in which answers are typed, then "Preview My Answers" does not handle answers which contain variables nicely. Instead the PG warning
```
The evaluated answer is not an answer hash : ||.
```
gets issued.

For example, in the attached question [matrixQ.pg.txt](https://github.com/openwebwork/pg/files/7697506/matrixQ.pg.txt) enter `x` into one of the input boxes and 0 in the rest. 

"Check answers" gives a more helpful feedback message like:
```
In entry (1,1): 	Your answer isn't a number
(it looks like a formula that returns a number)
```
(when the variable was in the 1,1 location in the matrix).

When an input box is left empty, preview does not trigger the warning message.

**Comment by dpvc** [comment]
This turns out to be due to the fact that preview mode doesn't produce the warning message about an entry being a formula not a number, and that another part of the code uses the presence of an error message to skip further checking.  Without the error message, code was run that didn't work when the student answer is a formula.

I have made a PR (#620) to resolve the issue.

**Comment by drgrice1** [comment]
I added the ArrayLayout back to htdocs/js/apps/Problem/problem.scss.

**Comment by taniwallach** [comment]
Resolved in PG-2.17 by https://github.com/openwebwork/pg/pull/620
